### ISSUE TITLE
Allow discarding unmerged changes (or auto-select appropriate option for merge on discard)

### ISSUE THREAD

**Issue by michael-hawker** [ForkIssues/TrackerWin#1859]
If I've got a merge conflict (from applying a stash for instance), I sometimes don't care about the changes to that file anymore and just want to get rid of it.

![image](https://user-images.githubusercontent.com/24302614/230498481-21b31a72-12d9-4ae0-87f1-0953035f8559.png)

If I right-click on it and do 'discard changes' though I get a git error:

> git checkout-index: components/Extensions/tests/Extensions.Tests.projitems is unmerged

![image](https://user-images.githubusercontent.com/24302614/230498546-64e6910c-83b5-45c6-ab31-7f61ca0f82b6.png)

As it's waiting for my resolution of the merge of this file, but I don't care. I just want it gone and to keep **whatever state I have in the branch**, that's my decision. _I don't want to have to think about which side of the file checkbox that is for like 20 files and click merge for each one individually_. I just want to select them all in the Unstaged view and 'discard' them.

Can this workflow pattern be optimized to make it smarter and do whatever git needs so that it doesn't just complain about this unmerged state? I don't want to repeat this process for the dozens of these I have in this case:

![image](https://user-images.githubusercontent.com/24302614/230499192-7e4ffd1a-ffa1-41c3-9add-6b45fe0dc48e.png)

i.e. just choosing discard on the file gets me here (in the example of the screenshots above):

![image](https://user-images.githubusercontent.com/24302614/230498687-46cfafbf-b92c-40c6-8327-4a9a4bcb2b58.png)

Or if I've selected multiples, all of them.

Thanks!

**Comment by DanPristupov** [comment]
> I sometimes don't care about the changes to that file anymore and just want to get rid of it.

It means, you want to select changes from the active branch which is 'llama/extensions-two`. The answer is on the first screenshot.

You can't discard changes because there are two of them. It's like answering 'no' to 'Do you want to stop or proceed'. What you are trying to say, is that you want changes like they were in the active branch before the merge'

> I just want to select them all in the Unstaged view and 'discard' them.

You can also select them all and select 'llama/extensions-two' in the context menu.

**Comment by rhys-vdw** [comment]
I would love this feature. Nothing more annoying than accidentally applying a stash and not being able to discard it afterwards because it has one or more conflicts that I never cared about.

> You can't discard changes because there are two of them. It's like answering 'no' to 'Do you want to stop or proceed'. What you are trying to say, is that you want changes like they were in the active branch before the merge'

This has not been a problem for other git GUIs. You just discard changes to reset the file back to the previous commit, regardless of whether changes are conflicting or not.

**Comment by wonkee-kim** [comment]
I would love this feature too. It is so painful when there are 100+ items that are unnecessary and just want to drop the changes.

**Comment by rhys-vdw** [comment]
Yeah it's horrible. You make one bad input and you're stuck with all this busywork before you can start working again.

Still easily the worst part of Fork.

**Comment by rhys-vdw** [comment]
@DanPristupov Please let me discard all of this with a single keystroke. It's horrible, something that could take 1 second takes 10 minutes.

<img width="666" height="721" alt="Image" src="https://github.com/user-attachments/assets/038b9496-e045-4fd1-8cb6-dbefc0949181" />

> You can't discard changes because there are two of them. It's like answering 'no' to 'Do you want to stop or proceed'. What you are trying to say, is that you want changes like they were in the active branch before the merge'

When you discard a conflicted file, the result should be that the changes are entirely gone. There is only one way to do that, which is to take the state of the file on the parent commit.
